packages/@aws-cdk/aws-codedeploy/lib/ecs/deployment-group.ts

```diff
@@ -55,8 +211,134 @@ export class EcsDeploymentGroup {
     return new ImportedEcsDeploymentGroup(scope, id, attrs);
   }
 
-  private constructor() {
-    // nothing to do until CFN supports ECS deployment groups
+  public readonly application: IEcsApplication;
+  public readonly deploymentGroupName: string;
+  public readonly deploymentGroupArn: string;
+  public readonly deploymentConfig: IEcsDeploymentConfig;
+  /**
+   * The service Role of this Deployment Group.
+   */
+  public readonly role: iam.IRole;
+
+  private readonly alarms: cloudwatch.IAlarm[];
+
+  constructor(scope: Construct, id: string, props: EcsDeploymentGroupProps) {
+    super(scope, id, {
+      physicalName: props.deploymentGroupName,
+    });
+
+    this.application = props.application || new EcsApplication(this, 'Application');
+    this.alarms = props.alarms || [];
+
+    this.role = props.role || new iam.Role(this, 'ServiceRole', {
+      assumedBy: new iam.ServicePrincipal('codedeploy.amazonaws.com'),
+    });
+
+    this.role.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('AWSCodeDeployRoleForECS'));
+    this.deploymentConfig = props.deploymentConfig || EcsDeploymentConfig.ALL_AT_ONCE;
+
+    for (const service of props.services) {
+      if (service instanceof ecs.BaseService) {
+        // Ensure the service's deployment controller is CodeDeploy
+        const cfnService = service.node.defaultChild as ecs.CfnService;
+        cfnService.deploymentController = { type: ecs.DeploymentControllerType.CODE_DEPLOY };
+
+        // Strip the revision ID from the service's task definition property to
+        // prevent new task def revisions in the stack from triggering updates
+        // to the stack's ECS service resource
+        cfnService.taskDefinition = service.taskDefinition.family;
```

Comment:
Just curious, what does ECS do with a task definition without a revision?

Reply:
It assumes `latest`. This allow the initial deploy to complete successfully but then also caused CloudFormation to ignore any new revisions of the TaskDefinition since we only want deployments to come through CodeDeploy and not CloudFormation.